src/client/runner/automation/playback/select/index.js

```diff
@@ -0,0 +1,218 @@
+
+        this.clientPoint = selectUtils.excludeElementScroll(this.element, point);
+
+        var moveArguments = {
+            element: document.documentElement,
+            offsetX: this.clientPoint.x,
+            offsetY: this.clientPoint.y
+        };
+
+        return this._move(moveArguments);
+    }
+
+    _mousedown () {
+        return cursor
+            .leftButtonDown()
+            .then(() => {
+                this.eventArgs = SelectAutomation._calculateEventArguments(this.clientPoint);
+
+                // NOTE: in WebKit and IE, the mousedown event opens the select element's dropdown.
+                // Therefore, we should prevent mousedown and hide the dropdown (B236416)
+                var needCloseSelectDropDown = (browserUtils.isWebKit || browserUtils.isIE) &&
+                                              domUtils.isSelectElement(this.element);
+
+                if (needCloseSelectDropDown)
+                    this._bindMousedownHandler();
+
+                if (browserUtils.hasTouchEvents) {
+                    this.eventState.simulateDefaultBehavior = eventSimulator.touchstart(this.eventArgs.element,
+                        this.eventArgs.options);
+                }
+                else {
+                    this.eventState.simulateDefaultBehavior = eventSimulator.mousedown(this.eventArgs.element,
+                        this.eventArgs.options);
+                }
+
+
+                if (this.eventState.simulateDefaultBehavior === false) {
+                    this.eventState.simulateDefaultBehavior = needCloseSelectDropDown &&
+                                                              !this.eventState.mousedownPrevented;
+                }
+
+                return this._focus();
+            });
+    }
+
+    _focus () {
+        return new Promise(resolve => {
+            // NOTE: If the target element is a child of a contentEditable element, we need to call focus for its parent
+            var elementForFocus = domUtils.isContentEditableElement(this.element) ?
+                                  contentEditable.findContentEditableParent(this.element) : this.element;
+
+            focusBlurSandbox.focus(elementForFocus, resolve, false, true);
+        });
+    }
+
+    _setSelection () {
+        var isTextEditable    = domUtils.isTextEditableElement(this.element);
+        var isContentEditable = domUtils.isContentEditableElement(this.element);
+
+        if (!(isTextEditable || isContentEditable) || this.eventState.simulateDefaultBehavior === false)
+            return;
+
+        // NOTE: The same cursor position may correspond to different nodes, so, if we
+        // know which nodes should be selected eventually, we should select them directly.
+        if (this.startNode) {
+            if (this.backward)
+                textSelection.selectByNodesAndOffsets(this.endNode, this.endOffset, this.startNode, this.startOffset, true, this.backward);
```

Comment:
There is `this.backward` duplication. We send arguments to `selectByNodesAndOffsets` in different order depending on `backward` and also send `backward` as the last argument. I suppose we can fix arguments order as `(start, end, ..., backward)`. The `selectByNodesAndOffsets` function should resolve arguments order inside them